<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="viewport" content="width=device-width, initial-scale=1">

<meta name="author" content="Anna Plantinga" />

<meta name="date" content="2016-11-22" />

<title>MiRKAT-S Package</title>



<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>



<link href="data:text/css;charset=utf-8,body%20%7B%0Abackground%2Dcolor%3A%20%23fff%3B%0Amargin%3A%201em%20auto%3B%0Amax%2Dwidth%3A%20700px%3B%0Aoverflow%3A%20visible%3B%0Apadding%2Dleft%3A%202em%3B%0Apadding%2Dright%3A%202em%3B%0Afont%2Dfamily%3A%20%22Open%20Sans%22%2C%20%22Helvetica%20Neue%22%2C%20Helvetica%2C%20Arial%2C%20sans%2Dserif%3B%0Afont%2Dsize%3A%2014px%3B%0Aline%2Dheight%3A%201%2E35%3B%0A%7D%0A%23header%20%7B%0Atext%2Dalign%3A%20center%3B%0A%7D%0A%23TOC%20%7B%0Aclear%3A%20both%3B%0Amargin%3A%200%200%2010px%2010px%3B%0Apadding%3A%204px%3B%0Awidth%3A%20400px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Aborder%2Dradius%3A%205px%3B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Afont%2Dsize%3A%2013px%3B%0Aline%2Dheight%3A%201%2E3%3B%0A%7D%0A%23TOC%20%2Etoctitle%20%7B%0Afont%2Dweight%3A%20bold%3B%0Afont%2Dsize%3A%2015px%3B%0Amargin%2Dleft%3A%205px%3B%0A%7D%0A%23TOC%20ul%20%7B%0Apadding%2Dleft%3A%2040px%3B%0Amargin%2Dleft%3A%20%2D1%2E5em%3B%0Amargin%2Dtop%3A%205px%3B%0Amargin%2Dbottom%3A%205px%3B%0A%7D%0A%23TOC%20ul%20ul%20%7B%0Amargin%2Dleft%3A%20%2D2em%3B%0A%7D%0A%23TOC%20li%20%7B%0Aline%2Dheight%3A%2016px%3B%0A%7D%0Atable%20%7B%0Amargin%3A%201em%20auto%3B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dcolor%3A%20%23DDDDDD%3B%0Aborder%2Dstyle%3A%20outset%3B%0Aborder%2Dcollapse%3A%20collapse%3B%0A%7D%0Atable%20th%20%7B%0Aborder%2Dwidth%3A%202px%3B%0Apadding%3A%205px%3B%0Aborder%2Dstyle%3A%20inset%3B%0A%7D%0Atable%20td%20%7B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dstyle%3A%20inset%3B%0Aline%2Dheight%3A%2018px%3B%0Apadding%3A%205px%205px%3B%0A%7D%0Atable%2C%20table%20th%2C%20table%20td%20%7B%0Aborder%2Dleft%2Dstyle%3A%20none%3B%0Aborder%2Dright%2Dstyle%3A%20none%3B%0A%7D%0Atable%20thead%2C%20table%20tr%2Eeven%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Ap%20%7B%0Amargin%3A%200%2E5em%200%3B%0A%7D%0Ablockquote%20%7B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Apadding%3A%200%2E25em%200%2E75em%3B%0A%7D%0Ahr%20%7B%0Aborder%2Dstyle%3A%20solid%3B%0Aborder%3A%20none%3B%0Aborder%2Dtop%3A%201px%20solid%20%23777%3B%0Amargin%3A%2028px%200%3B%0A%7D%0Adl%20%7B%0Amargin%2Dleft%3A%200%3B%0A%7D%0Adl%20dd%20%7B%0Amargin%2Dbottom%3A%2013px%3B%0Amargin%2Dleft%3A%2013px%3B%0A%7D%0Adl%20dt%20%7B%0Afont%2Dweight%3A%20bold%3B%0A%7D%0Aul%20%7B%0Amargin%2Dtop%3A%200%3B%0A%7D%0Aul%20li%20%7B%0Alist%2Dstyle%3A%20circle%20outside%3B%0A%7D%0Aul%20ul%20%7B%0Amargin%2Dbottom%3A%200%3B%0A%7D%0Apre%2C%20code%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0Aborder%2Dradius%3A%203px%3B%0Acolor%3A%20%23333%3B%0Awhite%2Dspace%3A%20pre%2Dwrap%3B%20%0A%7D%0Apre%20%7B%0Aborder%2Dradius%3A%203px%3B%0Amargin%3A%205px%200px%2010px%200px%3B%0Apadding%3A%2010px%3B%0A%7D%0Apre%3Anot%28%5Bclass%5D%29%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Acode%20%7B%0Afont%2Dfamily%3A%20Consolas%2C%20Monaco%2C%20%27Courier%20New%27%2C%20monospace%3B%0Afont%2Dsize%3A%2085%25%3B%0A%7D%0Ap%20%3E%20code%2C%20li%20%3E%20code%20%7B%0Apadding%3A%202px%200px%3B%0A%7D%0Adiv%2Efigure%20%7B%0Atext%2Dalign%3A%20center%3B%0A%7D%0Aimg%20%7B%0Abackground%2Dcolor%3A%20%23FFFFFF%3B%0Apadding%3A%202px%3B%0Aborder%3A%201px%20solid%20%23DDDDDD%3B%0Aborder%2Dradius%3A%203px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Amargin%3A%200%205px%3B%0A%7D%0Ah1%20%7B%0Amargin%2Dtop%3A%200%3B%0Afont%2Dsize%3A%2035px%3B%0Aline%2Dheight%3A%2040px%3B%0A%7D%0Ah2%20%7B%0Aborder%2Dbottom%3A%204px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Apadding%2Dbottom%3A%202px%3B%0Afont%2Dsize%3A%20145%25%3B%0A%7D%0Ah3%20%7B%0Aborder%2Dbottom%3A%202px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Afont%2Dsize%3A%20120%25%3B%0A%7D%0Ah4%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23f7f7f7%3B%0Amargin%2Dleft%3A%208px%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Ah5%2C%20h6%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23ccc%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Aa%20%7B%0Acolor%3A%20%230033dd%3B%0Atext%2Ddecoration%3A%20none%3B%0A%7D%0Aa%3Ahover%20%7B%0Acolor%3A%20%236666ff%3B%20%7D%0Aa%3Avisited%20%7B%0Acolor%3A%20%23800080%3B%20%7D%0Aa%3Avisited%3Ahover%20%7B%0Acolor%3A%20%23BB00BB%3B%20%7D%0Aa%5Bhref%5E%3D%22http%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0Aa%5Bhref%5E%3D%22https%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0A%0Acode%20%3E%20span%2Ekw%20%7B%20color%3A%20%23555%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Edt%20%7B%20color%3A%20%23902000%3B%20%7D%20%0Acode%20%3E%20span%2Edv%20%7B%20color%3A%20%2340a070%3B%20%7D%20%0Acode%20%3E%20span%2Ebn%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Efl%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Ech%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Est%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Eco%20%7B%20color%3A%20%23888888%3B%20font%2Dstyle%3A%20italic%3B%20%7D%20%0Acode%20%3E%20span%2Eot%20%7B%20color%3A%20%23007020%3B%20%7D%20%0Acode%20%3E%20span%2Eal%20%7B%20color%3A%20%23ff0000%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Efu%20%7B%20color%3A%20%23900%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%20code%20%3E%20span%2Eer%20%7B%20color%3A%20%23a61717%3B%20background%2Dcolor%3A%20%23e3d2d2%3B%20%7D%20%0A" rel="stylesheet" type="text/css" />

</head>

<body>




<h1 class="title toc-ignore">MiRKAT-S Package</h1>
<h4 class="author"><em>Anna Plantinga</em></h4>
<h4 class="date"><em>November 22, 2016</em></h4>



<div id="overview" class="section level2">
<h2>Overview</h2>
<p>The R package MiRKAT-S contains functions to test an association between the microbiota (taxonomic profiles of a microbial community) and suvival outcomes via an kernel matrix. The kernel matrix is based upon distance or dissimilarity metrics designed to summarize ecological and/or microbial communities, possibly incorporating phylogenetic information.</p>
</div>
<div id="required-packages" class="section level2">
<h2>Required packages</h2>
<p>CompQuadForm, survival, and GUniFrac packages are required for MiRKAT-S. All three required packages are available on CRAN (<a href="https://cran.r-project.org/web/packages/" class="uri">https://cran.r-project.org/web/packages/</a>). In this vignette, we use the throat data from GUniFrac for demonstration. In our demonstration, kernels are constructed from the family of UniFrac distances and the Bray-Curtis dissimilarity; however, additional options for relevant kernels are mentioned.</p>
</div>
<div id="using-mirkat-s-to-test-associations-between-microbiota-and-survival" class="section level2">
<h2>Using MiRKAT-S to test associations between microbiota and survival</h2>
<div id="load-dataset-and-prepare-data" class="section level3">
<h3>Load dataset and prepare data</h3>
<p>We use the throat microbiome data from package GUniFrac to demonstrate the use of MiRKAT-S. These data were generated to study the effect of smoking on the microbiota of the upper respiratory tract in 60 individuals, 28 smokers and 32 nonsmokers. As such, the original dataset includes samples from the microbiome of the nasopharynx and oropharynx on each side of the body. The dataset used here consists of the oropharynx samples from the left side of the body.</p>
<p>Because the dataset has a binary phenotype (smoking) rather than a measure of censored time to event outcomes, we consider smoking status and gender as covariates and generate null outcome data from the Exponential distribution. Specifically, we generate survival times as <span class="math inline">\(S \sim \text{Exponential}(1 + I(\text{smoke}) + I(\text{male}))\)</span>, and censoring times as <span class="math inline">\(C \sim \text{Exponential}(0.75)\)</span>. Then the outcome measure consists of <span class="math inline">\(T = \text{min}(S, C)\)</span> and <span class="math inline">\(\Delta = I(S \leq C)\)</span>. This simulation procedure results in approximately 33% censoring.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Load the data </span>
<span class="kw">library</span>(MiRKATS, <span class="dt">quietly=</span><span class="ot">TRUE</span>)</code></pre></div>
<pre><code>## This is vegan 2.4-1</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(GUniFrac, <span class="dt">quietly=</span><span class="ot">TRUE</span>)
<span class="kw">data</span>(throat.tree)
<span class="kw">data</span>(throat.otu.tab)
<span class="kw">data</span>(throat.meta)

<span class="co"># Prepare covariates </span>
<span class="kw">set.seed</span>(<span class="dv">1</span>)
Male =<span class="st"> </span><span class="kw">rep</span>(<span class="dv">0</span>, <span class="kw">nrow</span>(throat.meta))
Male[throat.meta$Sex ==<span class="st"> &quot;Male&quot;</span>] &lt;-<span class="st"> </span><span class="dv">1</span>
Smoker =<span class="st"> </span><span class="kw">rep</span>(<span class="dv">0</span>, <span class="kw">nrow</span>(throat.meta)) 
Smoker[throat.meta$SmokingStatus ==<span class="st"> &quot;Smoker&quot;</span>] &lt;-<span class="st"> </span><span class="dv">1</span> 

<span class="co"># Simulate outcomes </span>
<span class="co"># Here, outcome is associated with covariates but unassociated with microbiota</span>
<span class="co"># 33% censoring </span>
SurvTime &lt;-<span class="st"> </span><span class="kw">rexp</span>(<span class="dv">60</span>, (<span class="dv">1</span> +<span class="st"> </span>Smoker +<span class="st"> </span>Male))
CensTime &lt;-<span class="st"> </span><span class="kw">rexp</span>(<span class="dv">60</span>, <span class="fl">0.75</span>)
Delta &lt;-<span class="st"> </span><span class="kw">as.numeric</span>( SurvTime &lt;=<span class="st"> </span>CensTime )
ObsTime &lt;-<span class="st"> </span><span class="kw">pmin</span>(SurvTime, CensTime)</code></pre></div>
</div>
<div id="create-distance-matrices" class="section level3">
<h3>Create distance matrices</h3>
<p>Many options exist for distance or dissimilarity metrics. Below, we calculate the distances used in the manuscript: the unweighted and weighted UniFrac distances, the generalized UniFrac distance with <span class="math inline">\(\alpha=0.5\)</span>, and the Bray-Curtis dissimilarity. Choice of distance metric (and therefore kernel) is discussed following the example.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">unifracs &lt;-<span class="st"> </span><span class="kw">GUniFrac</span>(throat.otu.tab, throat.tree, <span class="dt">alpha=</span><span class="kw">c</span>(<span class="dv">0</span>, <span class="fl">0.5</span>, <span class="dv">1</span>))$unifracs
D.weighted &lt;-<span class="st"> </span>unifracs[,,<span class="st">&quot;d_1&quot;</span>]  
D.unweighted &lt;-<span class="st"> </span>unifracs[,,<span class="st">&quot;d_UW&quot;</span>]
D.generalized &lt;-<span class="st"> </span>unifracs[,,<span class="st">&quot;d_0.5&quot;</span>]
D.braycurtis &lt;-<span class="st"> </span><span class="kw">as.matrix</span>(<span class="kw">vegdist</span>(throat.otu.tab, <span class="dt">method=</span><span class="st">&quot;bray&quot;</span>))</code></pre></div>
</div>
<div id="convert-distance-matrices-to-kernel-matrices" class="section level3">
<h3>Convert distance matrices to kernel matrices</h3>
<p>The D2K function in MiRKAT-S converts distance matrices to kernel matrices via <span class="math display">\[
K = -\frac{1}{2} \left(I - \frac{1 1'}{n}\right) D^2 \left(I - \frac{1 1'}{n}\right).
\]</span> Here, <span class="math inline">\(I\)</span> is the identity matrix and <span class="math inline">\(1\)</span> is an <span class="math inline">\(n\)</span>-vector of ones. To ensure that <span class="math inline">\(K\)</span> is positive semi-definite, we replace negative eigenvalues with zero. That is, we perform an eigenvalue decomposition <span class="math inline">\(K = U \Lambda U\)</span>, where <span class="math inline">\(\Lambda = \text{diag}(\lambda_1,...,\lambda_n)\)</span>, and then reconstruct the kernel matrix using the nonnegative eigenvalues <span class="math inline">\(\Lambda^* = \text{diag}(\text{max}(\lambda_1,0),...,\text{max}(\lambda_n,0))\)</span> so that <span class="math inline">\(K = U \Lambda^* U\)</span>.</p>
<p>Alternatively, a distance matrix can be used as input to MiRKATS (argument kd) with argument distance=TRUE; the distance will be automatically converted to a kernel matrix exactly as described above.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">K.weighted &lt;-<span class="st"> </span><span class="kw">D2K</span>(D.weighted) 
K.unweighted &lt;-<span class="st"> </span><span class="kw">D2K</span>(D.unweighted)
K.generalized &lt;-<span class="st"> </span><span class="kw">D2K</span>(D.generalized) 
K.braycurtis &lt;-<span class="st"> </span><span class="kw">D2K</span>(D.braycurtis)</code></pre></div>
</div>
<div id="test-using-a-single-kernel-mirkat-s-p-values" class="section level3">
<h3>Test using a single kernel, MiRKAT-S p-values</h3>
<p>Here, we run MiRKAT-S to test the association between the microbiota and simulated survival times, adjusting for gender and smoking status.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># use kernel matrix with distance=FALSE </span>
<span class="kw">MiRKATS</span>(<span class="dt">kd =</span> K.generalized, <span class="dt">distance =</span> <span class="ot">FALSE</span>, <span class="dt">obstime =</span> ObsTime, <span class="dt">delta =</span> Delta, <span class="dt">covar =</span> <span class="kw">cbind</span>(Male, Smoker))</code></pre></div>
<pre><code>## [1] 0.08217796</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># equivalently, use distance matrix with distance=TRUE </span>
<span class="kw">MiRKATS</span>(<span class="dt">kd =</span> D.generalized, <span class="dt">distance =</span> <span class="ot">TRUE</span>, <span class="dt">obstime =</span> ObsTime, <span class="dt">delta =</span> Delta, <span class="dt">covar =</span> <span class="kw">cbind</span>(Male, Smoker))</code></pre></div>
<pre><code>## [1] 0.08217796</code></pre>
<p>The argument “distance” indicates whether “kd” is a distance matrix (TRUE) or kernel matrix (FALSE). The output is the p-value for the test using Davies’ exact method, which computes the p-value based on a mixture of chi-square distributions. We use a small-sample correction to account for the modest sample sizes and sparse OTU count matrices that often result from studies of the microbiome.</p>
</div>
<div id="test-using-a-single-kernel-permutation-p-values" class="section level3">
<h3>Test using a single kernel, permutation p-values</h3>
<p>Because there are only 60 individuals in this dataset, permutation is a reasonable choice for p-value calculation, as MiRKAT-S may be slightly anti-conservative with very small samples. MiRKAT-S will generate a warning when permutation is not used for sample sizes <span class="math inline">\(n \leq 50\)</span>. “nperm” indicates the number of permutations to perform to generate the p-value (default = 1000).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">MiRKATS</span>(<span class="dt">kd =</span> K.generalized, <span class="dt">distance =</span> <span class="ot">FALSE</span>, <span class="dt">obstime =</span> ObsTime, <span class="dt">delta =</span> Delta, <span class="dt">covar =</span> <span class="kw">cbind</span>(Male, Smoker), <span class="dt">perm=</span><span class="ot">TRUE</span>, <span class="dt">nperm=</span><span class="dv">1000</span>)</code></pre></div>
<pre><code>## [1] 0.12</code></pre>
</div>
</div>
<div id="selection-of-kernels-for-testing" class="section level2">
<h2>Selection of kernel(s) for testing</h2>
<p>How to choose an appropriate distance matrix and kernel for testing is a difficult question. However, it is important, since the distance metric used to generate the kernel for MiRKAT-S strongly affects the power of the test. In particular, MiRKAT-S has highest power when the form of association between the microbiota and the outcome assumed by the kernel matches the true form of association.</p>
<p>In the case of the UniFrac families and the Bray-Curtis dissimilarity, the factors at play are (1) the abundance of the associated taxa and (2) whether closely related taxa (phylogenetically) tend to be related or not related to the outcome as a group. For example, the following are some of the distance metrics that have been used for studies of the microbiome:</p>
<table style="width:104%;">
<colgroup>
<col width="23%"></col>
<col width="15%"></col>
<col width="15%"></col>
<col width="38%"></col>
<col width="11%"></col>
</colgroup>
<thead>
<tr class="header">
<th align="left">Distance</th>
<th align="center">Phylogeny?</th>
<th align="center">Abundance?</th>
<th align="left">Other notes</th>
<th align="center">Reference</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Unweighted UniFrac</td>
<td align="center">√</td>
<td align="center">X</td>
<td align="left"></td>
<td align="center">1</td>
</tr>
<tr class="even">
<td align="left">Weighted UniFrac</td>
<td align="center">√</td>
<td align="center">√</td>
<td align="left"></td>
<td align="center">2</td>
</tr>
<tr class="odd">
<td align="left">Generalized UniFrac</td>
<td align="center">√ </td>
<td align="center">(√)</td>
<td align="left">Parameter alpha defines extent to which abundance is taken into account</td>
<td align="center">3</td>
</tr>
<tr class="even">
<td align="left">Jaccard</td>
<td align="center">X</td>
<td align="center">X</td>
<td align="left">1 - (taxa in both)/(taxa in either); typically presence/absence, but can be extended to an abundance-weighted version</td>
<td align="center">4,5</td>
</tr>
<tr class="odd">
<td align="left">Bray-Curtis</td>
<td align="center">X</td>
<td align="center">√ </td>
<td align="left">Similar to Jaccard, but uses counts</td>
<td align="center">6</td>
</tr>
</tbody>
</table>
<p>In the table above, √ indicates the distance or dissimilarity metric has the feature; (√) indicates that the feature is present either in some variations of the metric or is present to some extent; and X indicates that the feature is not present.</p>
<p>Depending on which of these characteristics are expected to be present in a particular study (based on prior knowledge or intuition), an appropriate distance or dissimilarity can be selected. If the study is exploratory and strong protection of type 1 error is not needed, several distance metrics can be explored. Depending on which one(s) are highly significant, some information can be gained about the nature of any association between the microbiota and the outcome.</p>
</div>
<div id="references" class="section level2">
<h2>References</h2>
<div id="distances-and-dissimilarities" class="section level3">
<h3>Distances and dissimilarities</h3>
<ol style="list-style-type: decimal">
<li>Lozupone C &amp; Knight R (2005). UniFrac: a new phylogenetic method for comparing microbial communities. Applied and Environmental Microbiology, 71(12), 8228-8235.</li>
<li>Lozupone CA, Hamady M, Kelley ST, &amp; Knight, R (2007). Quantitative and qualitative β diversity measures lead to different insights into factors that structure microbial communities. Applied and Environmental Microbiology, 73(5), 1576-1585.</li>
<li>Chen J, Bittinger K, Charlson ES, Hoffmann C, Lewis J, Wu GD, Collman RG, Bushman FD, &amp; Li H (2012). Associating microbiome composition with environmental covariates using generalized UniFrac distances. Bioinformatics, 28(16), 2106-2113.</li>
<li>Levandowsky M &amp; Winter D (1971). Distance between sets. Nature, 234(5323), 34-35.</li>
<li>Chao A, Chazdon RL, Colwell RK, &amp; Shen TJ (2005). A new statistical approach for assessing similarity of species composition with incidence and abundance data. Ecology Letters, 8(2), 148-159.</li>
<li>Bray JR &amp; Curtis JT (1957). An ordination of the upland forest communities of southern Wisconsin. Ecological Monographs, 27(4), 325-349.</li>
</ol>
</div>
<div id="mirkat-s-method" class="section level3">
<h3>MiRKAT-S method</h3>
<ol style="list-style-type: decimal">
<li>Plantinga A, Zhan X, Zhao N, Chen J, Jenq RR, Wu MC (2016). MiRKAT-S: A community-level test of association between taxonomic profiles and survival times. Under revision.</li>
<li>Zhao N, Chen J, Carroll IM, Ringel-Kulka T, Epstein MP, Zhou H, Zhou JJ, Ringel Y, Li H, Wu MC (2015). Testing in microbiome-profiling studies with MiRKAT, the microbiome regression-based kernel association test. The American Journal of Human Genetics, 96(5), 797-807.</li>
<li>Davies RB (1980). The distribution of a linear combination of chi-2 random variables. Journal of the Royal Statistical Society Series C (Applied Statistics), 29(3), 323-333.</li>
<li>Chen J, Chen W, Zhao N, Wu MC, Schaid DJ (2016). Small sample kernel association tests for human genetic and microbiome association studies. Genetic Epidemiology, 40(1), 5-19.</li>
</ol>
</div>
</div>



<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
